src/subscription.rs

```diff
@@ -132,3 +143,88 @@ impl<'a, T: Runtime> EventSubscription<'a, T> {
         }
     }
 }
+
+/// Stream storage set changes
+pub trait StorageChangeStream {
+    /// Type to return from next.
+    type Item;
+
+    /// Fetch the next storage item.
+    fn next<'a>(
+        &'a mut self,
+    ) -> Pin<Box<dyn Future<Output = Option<Self::Item>> + Send + 'a>>;
+}
```

Comment:
What's the story with the `futures` `Stream` trait nowadays? I did try using it in a similar context a long time ago but found the async interop tricky. Maybe it's improved since then.

Reply:
It's not async AFAIK, which makes writing custom hooks trickier.